### DISCUSSION TITLE
How do I terminate a running dev container?

### DISCUSSION THREAD

**Discussion by reynoldsbd** [devcontainers/spec#139] [Q&A] [answered]
The CLI includes a `devcontainer up` command that can be used to start a running container. But how can I terminate that container once I'm done with it?

Seems like there should be a `devcontainer down` command or something similar? Or perhaps I am misunderstanding something about the devcontainer lifecycle

**Comment by samruddhikhandale** [comment, marked as answer]
Hi 👋  

The dev container CLI is currently in active development, and it consists of the following commands mentioned in [here](https://github.com/devcontainers/cli/blob/main/README.md#context). It mentions the `devcontainer stop` & `devcontainer down` commands. However, they haven't been implemented yet // cc @bamurtaugh 

In the meanwhile, you could clean up the dev container as follows -
- `id_label="test-container=containerX"`
- `devcontainer up --id-label ${id_label} --workspace-folder ../workspace` --> Create & run dev container
- `devcontainer exec --id-label ${id_label} --workspace-folder ../workspace bash -c '....'` --> Execute a command on a running dev container
- `docker rm -f $(docker container ls -f "label=${id_label}" -q)` -->  ❗ **Remove dev container** ❗

**Reply by 8ware** [reply]
`devcontainer exec --workspace-folder ../workspace hostname | xargs -r docker rm -f` won't even require the `${id_label}`.

**Comment by Jarrodsz** [comment]
would it be possible to create a stop.sh script that incorperates this? perhaps an example would be great in the meantime untill those down and stop commands are in place?
